src/post/post.entity.ts

```diff
@@ -338,6 +338,11 @@ export class Post extends BaseEntity {
         );
 
         for (const mention of data.mentions ?? []) {
+            if (data.inReplyTo && mention.id === data.inReplyTo.author.id) {
```

Comment:
My concern with this approach is that we won't save the mentions in db. If someone mentions the author in a reply we won't have that mention saved in the db(although it's a valid mention) 

I agree with fabien that this should be handled on the notification level.